src/js/components/Select/SelectContainer.js

```diff
@@ -109,11 +105,14 @@ const SelectContainer = forwardRef(
       // need to wait for Drop to be ready
       const timer = setTimeout(() => {
         const optionsNode = optionsRef.current;
+        const clearButton = clearRef.current;
         if (onSearch) {
           const searchInput = searchRef.current;
           if (searchInput && searchInput.focus) {
             setFocusWithoutScroll(searchInput);
           }
+        } else if (clear && clearButton && clearButton.focus) {
```

Comment:
I think we should only set focus on the clear button if it is at the top. If `clear.position = "bottom"` we should just set focus on the first option.